node/node.go

```diff
@@ -1248,13 +1266,16 @@ func NewNode(config *cfg.Config,
 	)
 
 	if config.P2P.PexReactor {
-		pexReactor = createPEXReactorAndAddToSwitch(addrBook, config, sw, logger)
-		pexReactorV2, err = createPEXReactorV2(config, logger, peerManager, router)
-		if err != nil {
-			return nil, err
+		router.AddChannelDescriptors(pex.ChannelDescriptors)
+		transport.AddChannelDescriptors(pex.ChannelDescriptors)
+		if config.P2P.DisableLegacy {
+			pexReactorV2, err = createPEXReactorV2(config, logger, peerManager, router)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			pexReactor = createPEXReactorAndAddToSwitch(addrBook, config, sw, logger)
```

Comment:
maybe it makes sense to pass the channel descriptors into the constructor functions for the reactors?

Reply:
but the reactor itself doesn't really care which channel it's operating over as long as you give it a channel. The channel descriptors just need to be passed into the router (and currently the transport).